### PR TITLE
Added Quandl API

### PR DESCRIPTION
**Pull request by adrianyi**
Replaced pulling stock data from file to Quandl API.
It will require `QUANDL_KEY` environment variable.

Currently, the app calls the function with default values. It pulls 'open' price for 'GOOGL' for last 100 days.

### CODE REVIEW

File: stockticker/stockticker.py [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
     """create a flask app"""
     app = Flask(__name__)
 
     @app.route('/')
     def index(): # pylint: disable=W0612
         """main route"""
-        # Replace with API call to quandle
-        df = pd.read_csv('static/GOOGL_data.txt') # pylint: disable=C0103
+        # Add user input?
+        df = query_quandl('GOOGL', quandl_key, 'open', 100) # pylint: disable=C0103
```
Comment (ZachGlassman) [inline stockticker/stockticker.py]: we want this on user input, not when page loads

File: stockticker/stockticker.py [hunk 2]
```diff
@@ -1,21 +1,47 @@
...
+    # prepare date string for the Quandl API
+    today = datetime.today()
+    dates = ','.join((today-timedelta(days=i)).strftime('%Y-%m-%d') for i in range(1,days+1))
+    params = {'ticker': ticker, 'date': dates, 'api_key': quandl_key}
+    try:
+        r = requests.get('https://www.quandl.com/api/v3/datatables/WIKI/PRICES.json',
+                         params=params, timeout=10)
+    except Timeout:
+        raise QuandlException('Request timed out')
+    if r.status_code != 200:
```
Comment (ZachGlassman) [inline stockticker/stockticker.py]: look up `r.ok`, this might be better.

**Comment by ZachGlassman** [comment]
fixes https://github.com/thedataincubator/stockpredict/issues/5

**Comment by ZachGlassman** [comment]
a few `pylint` things left.  You can install `pylint` and run the tests locally if you want

**Comment by ZachGlassman** [comment]
LGTM, merge if @andreaslowe agrees.

I have added a few comments, you can make changes if you want.

**Review by ZachGlassman — APPROVED** [review]

File: stockticker/stockticker.py [hunk 3]
```diff
@@ -1,20 +1,48 @@
...
+    """Exception for Quandl API"""
+    pass
+
+def query_quandl(ticker, quandl_key, value='open', days=100):
+    """get stock value from quandl"""
+    days = min(days, 800) # getting much larger than this causes too long URI
+
+    # prepare date string for the Quandl API
+    today = datetime.today()
+    dates = ','.join((today-timedelta(days=i)).strftime('%Y-%m-%d') for i in range(1, days+1))
```
Comment (ZachGlassman) [inline stockticker/stockticker.py]: is there a way to specify `start_date` and `end_date` instead?

File: stockticker/stockticker.py [hunk 4]
```diff
@@ -1,20 +1,48 @@
...
+    try:
+        r = requests.get('https://www.quandl.com/api/v3/datatables/WIKI/PRICES.json', # pylint: disable=C0103
+                         params=params, timeout=10)
+    except Timeout:
+        raise QuandlException('Request timed out')
+    if not r.ok:
+        raise QuandlException('Request failed with status code {}'.format(r.status_code))
+    # we can catch more exceptions below, like JSONDecodeError, KeyError, etc.
+    raw = json.loads(r.text)
+    # raw['datatable']['data'] = [['GOOGL', '2018-03-22', 1080.01, 1083.92, ...], ...]
```
Comment (ZachGlassman) [inline stockticker/stockticker.py]: delete this line, no point in having it here

File: stockticker/stockticker.py (hunk 3 above)
Comment (adrianyi) [reply stockticker/stockticker.py]: ah, I just learned about the different API call for this... I'll make the changes! It'll make cleaner.

**Review by andreaslowe — APPROVED** [review]

**Pull request merged**